File: flame/src/main/java/org/movieos/flame/MyApplication.java
package org.movieos.flame;

import android.app.Application;
import android.support.annotation.NonNull;

import org.movieos.flame.utilities.DiscoveryService;
import timber.log.Timber;

public class MyApplication extends Application {

    private static DiscoveryService sDiscoveryService;

    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }

        //noinspection AssignmentToStaticFieldFromInstanceMethod
        sDiscoveryService = new DiscoveryService(this);
    }

    @NonNull
    public static DiscoveryService discoveryService() {
        return sDiscoveryService;
    }
}

File: flame/src/main/java/org/movieos/flame/MyApplication.kt
package org.movieos.flame

import android.app.Application
import android.util.Log
import org.movieos.flame.utilities.DiscoveryService
import timber.log.Timber

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        singleton = this

        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())

        } else {
            Timber.plant(object : Timber.Tree() {
                override fun log(priority: Int, tag: String?, message: String?, throwable: Throwable?) {
                    if (priority == Log.VERBOSE || priority == Log.DEBUG || priority == Log.INFO) {
                        return
                    }
                }
            })
        }

    }

    companion object {
        var singleton: MyApplication? = null
        val discoveryService: DiscoveryService by lazy { DiscoveryService(singleton!!) }
    }
}
